src/test/java/de/uhd/ifi/se/decision/management/jira/classification/TestOnlineFileTrainerImpl.java

```diff
@@ -1,168 +1,171 @@
-        sentence.setRelevant(true);
-        sentence.setType(KnowledgeType.ALTERNATIVE);
-        sentence.setValidated(true);
-
-        assertTrue(trainer.update(sentence));
-    }
-
-    //TODO: tests with unvalidated data element
-
-    @Test
-    @NonTransactional
-    public void testGetClassifier() {
-        assertNotNull(this.trainer.getClassifier());
-    }
-
-    @Test
-    @NonTransactional
-    public void testGetInstances() {
-        assertNotNull(this.trainer.getInstances());
-    }
-
-    @Test
-    @NonTransactional
-    public void testMakeBinaryPredicition() {
-        assertEquals(2, this.trainer.getClassifier().makeBinaryPredictions(TEST_SENTENCES).size());
-    }
-
-    @Test
-    @NonTransactional
-    public void testMakeFineGrainedPredicition() {
-        assertEquals(2, this.trainer.getClassifier().makeFineGrainedPredictions(TEST_SENTENCES).size());
-    }
-
-    private File getTrimmedDefaultArffFile() {
-        File fullDefaultFile = new File("src/main/resources/classifier/defaultTrainingData.arff");
-        File trimmedDefaultFile = new File(TestFileTrainer.TEST_ARFF_FILE_PATH);
-
-        int numberOfLines = 25;
-
-        BufferedWriter writer = null;
-        try {
-            BufferedReader reader = new BufferedReader(new FileReader(fullDefaultFile));
-            writer = new BufferedWriter(new FileWriter(trimmedDefaultFile));
-
-            String currentLine;
-            int counter = 0;
-            while ((currentLine = reader.readLine()) != null && counter < numberOfLines) {
-                writer.write(currentLine + System.getProperty("line.separator"));
-                counter++;
-            }
-            writer.close();
-            reader.close();
-            return trimmedDefaultFile;
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        return fullDefaultFile;
-    }
-
+	private OnlineFileTrainerImpl trainer;
+	private static final List<String> TEST_SENTENCES = Arrays.asList("Pizza is preferred", "I have an issue");
+
+	@Before
+	public void setUp() {
+		init();
+		trainer = new OnlineFileTrainerImpl("TEST");
+		trainer.setTrainingData(getTrainingData());
+	}
+
+	@Test
+	@NonTransactional
+	public void testOnlineClassificationTrainerSetTrainingData() {
+		List<KnowledgeElement> trainingElements = getTrainingData();
+		trainer.setTrainingData(trainingElements);
+		assertTrue(trainer.train());
+	}
+
+	@Test
+	@NonTransactional
+	public void testOnlineClassificationTrainerFromArffFile() {
+		File file = trainer.saveTrainingFile(true);
+		trainer.setTrainingFile(file);
+		assertNotNull(trainer.getInstances());
+		trainer = new OnlineFileTrainerImpl("TEST", file.getName());
+		// assertNotNull(trainer.getInstances());
+		assertTrue(trainer.train());
+		file.delete();
+	}
+
+	@Test
+	@NonTransactional
+	public void testSaveArffFile() {
+		File file = trainer.saveTrainingFile(false);
+		Assert.assertTrue(file.exists());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'junit.framework.Assert.assertTrue'](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5419462)